k8s/aws/ebs-volumes/create-ebs-volume.yml

```diff
@@ -72,6 +73,15 @@
             mode: 0755
           with_items: "{{volumes.results}}"
 
+        - name: Creating a file for store disk-map information
+          lineinfile:
+            create: yes
+            state: present
+            path: "/tmp/aws/disk-map"
+            line: '{{ item.item.private_dns_name }},{{ item.volume_id }}'
```

Comment:
Just checking if this item.item.private_dns_name is required?

Reply:
yes! it is required because json is like this:
```
{
    "changed": true, 
    "device": "/dev/xvdb", 
    "invocation": {
        "module_args": {
            "aws_access_key": null, 
            "aws_secret_key": null, 
            "delete_on_termination": false, 
            "device_name": "/dev/xvdb", 
            "ec2_url": null, 
            "encrypted": false, 
            "id": null, 
            "instance": "i-02d12407ed19f4c4c", 
            "iops": null, 
            "kms_key_id": null, 
            "name": null, 
            "profile": null, 
            "region": "eu-west-2", 
            "security_token": null, 
            "snapshot": null, 
            "state": "present", 
            "tags": {}, 
            "validate_certs": true, 
            "volume_size": "50", 
            "volume_type": "gp2", 
            "zone": "eu-west-2a"
        }
    }, 
    "item": {
        "ami_launch_index": 0, 
        "architecture": "x86_64", 
        "block_device_mappings": [
            {
                "device_name": "/dev/sda1", 
                "ebs": {
                    "attach_time": "2018-12-14T10:23:12+00:00", 
                    "delete_on_termination": true, 
                    "status": "attached", 
                    "volume_id": "vol-0c559d14a47fcac14"
                }
            }
        ], 
        "client_token": "34e59eb0-f72b-98b2-7cb2-10c426ef1d76_subnet-06e31f54af6444209_3", 
        "cpu_options": {
            "core_count": 4, 
            "threads_per_core": 1
        }, 
        "ebs_optimized": false, 
        "ena_support": true, 
        "hypervisor": "xen", 
        "iam_instance_profile": {
            "arn": "arn:aws:iam::936055414837:instance-profile/nodes.k8s-chandan-test.k8s.local", 
            "id": "AIPAJQ5VKFOZ3GNPHTSFA"
        }, 
        "image_id": "ami-6b3fd60c", 
        "instance_id": "i-02d12407ed19f4c4c", 
        "instance_type": "t2.xlarge", 
        "key_name": "kubernetes.k8s-chandan-test.k8s.local-a5:d9:f7:5a:9d:80:8d:f8:60:df:73:cd:18:1d:61:c8", 
        "launch_time": "2018-12-14T10:23:11+00:00", 
        "monitoring": {
            "state": "disabled"
        }, 
        "network_interfaces": [
            {
                "association": {
                    "ip_owner_id": "amazon", 
                    "public_dns_name": "ec2-3-8-148-145.eu-west-2.compute.amazonaws.com", 
                    "public_ip": "3.8.148.145"
                }, 
                "attachment": {
                    "attach_time": "2018-12-14T10:23:11+00:00", 
                    "attachment_id": "eni-attach-08510cc0353f4cabe", 
                    "delete_on_termination": true, 
                    "device_index": 0, 
                    "status": "attached"
                }, 
                "description": "", 
                "groups": [
                    {
                        "group_id": "sg-0a089f36da9b99606", 
                        "group_name": "nodes.k8s-chandan-test.k8s.local"
                    }
                ], 
                "ipv6_addresses": [], 
                "mac_address": "06:81:c5:d1:a3:ac", 
                "network_interface_id": "eni-06cdba78d99b9be40", 
                "owner_id": "936055414837", 
                "private_dns_name": "ip-10-0-1-237.eu-west-2.compute.internal", 
                "private_ip_address": "10.0.1.237", 
                "private_ip_addresses": [
                    {
                        "association": {
                            "ip_owner_id": "amazon", 
                            "public_dns_name": "ec2-3-8-148-145.eu-west-2.compute.amazonaws.com", 
                            "public_ip": "3.8.148.145"
                        }, 
                        "primary": true, 
                        "private_dns_name": "ip-10-0-1-237.eu-west-2.compute.internal", 
                        "private_ip_address": "10.0.1.237"
                    }
                ], 
                "source_dest_check": false, 
                "status": "in-use", 
                "subnet_id": "subnet-06e31f54af6444209", 
                "vpc_id": "vpc-06ae9997b3ded5d4d"
            }
        ], 
        "placement": {
            "availability_zone": "eu-west-2a", 
            "group_name": "", 
            "tenancy": "default"
        }, 
        "private_dns_name": "ip-10-0-1-237.eu-west-2.compute.internal", 
        "private_ip_address": "10.0.1.237", 
        "product_codes": [], 
        "public_dns_name": "ec2-3-8-148-145.eu-west-2.compute.amazonaws.com", 
        "public_ip_address": "3.8.148.145", 
        "root_device_name": "/dev/sda1", 
        "root_device_type": "ebs", 
        "security_groups": [
            {
                "group_id": "sg-0a089f36da9b99606", 
                "group_name": "nodes.k8s-chandan-test.k8s.local"
            }
        ], 
        "source_dest_check": false, 
        "state": {
            "code": 16, 
            "name": "running"
        }, 
        "state_transition_reason": "", 
        "subnet_id": "subnet-06e31f54af6444209", 
        "tags": {
            "KubernetesCluster": "k8s-chandan-test.k8s.local", 
            "Name": "nodes.k8s-chandan-test.k8s.local", 
            "aws:autoscaling:groupName": "nodes.k8s-chandan-test.k8s.local", 
            "k8s.io/cluster-autoscaler/node-template/label/kops.k8s.io/instancegroup": "nodes", 
            "k8s.io/role/node": "1"
        }, 
        "virtualization_type": "hvm", 
        "vpc_id": "vpc-06ae9997b3ded5d4d"
    }, 
    "volume": {
        "attachment_set": {
            "attach_time": "2018-12-14T10:30:30.000Z", 
            "deleteOnTermination": "false", 
            "device": "/dev/xvdb", 
            "instance_id": "i-02d12407ed19f4c4c", 
            "status": "attached"
        }, 
        "create_time": "2018-12-14T10:30:23.906Z", 
        "encrypted": false, 
        "id": "vol-00b4539229b6d9887", 
        "iops": 150, 
        "size": 50, 
        "snapshot_id": "", 
        "status": "in-use", 
        "tags": {}, 
        "type": "gp2", 
        "zone": "eu-west-2a"
    }, 
    "volume_id": "vol-00b4539229b6d9887", 
    "volume_type": "gp2"
}

```

Reply:
First item is for iteration part and second item comes from json